pay-api/tests/conftest.py

```diff
@@ -67,160 +69,63 @@ def client_ctx(app):
         yield _client
 
 
-@pytest.fixture(scope='session')
+@pytest.fixture(scope='session', autouse=True)
 def db(app):  # pylint: disable=redefined-outer-name, invalid-name
-    """Return a session-wide initialised database.
-
-    Drops all existing tables - Meta follows Postgres FKs
-    """
+    """Return a session-wide initialised database."""
     with app.app_context():
-        # Clear out views
-        view_sql = """SELECT table_name FROM information_schema.views
-                WHERE table_schema='public'
-            """
-
-        sess = _db.session()
-        for seq in [name for (name,) in sess.execute(text(view_sql))]:
-            try:
-                sess.execute(text('DROP VIEW public.%s ;' % seq))
-                print('DROP VIEW public.%s ' % seq)
-            except Exception as err:  # NOQA pylint: disable=broad-except
-                print(f'Error: {err}')
-        sess.commit()
-
-        # Clear out any existing tables
-        metadata = MetaData(_db.engine)
-        metadata.reflect()
-        for table in metadata.tables.values():
-            for fk in table.foreign_keys:  # pylint: disable=invalid-name
-                _db.engine.execute(DropConstraint(fk.constraint))
-        metadata.drop_all()
-        _db.drop_all()
-
-        sequence_sql = """SELECT sequence_name FROM information_schema.sequences
-                          WHERE sequence_schema='public'
-                       """
-
-        sess = _db.session()
-        for seq in [name for (name,) in sess.execute(text(sequence_sql))]:
-            try:
-                sess.execute(text('DROP SEQUENCE public.%s ;' % seq))
-                print('DROP SEQUENCE public.%s ' % seq)
-            except Exception as err:  # NOQA pylint: disable=broad-except
-                print(f'Error: {err}')
-        sess.commit()
-
-        # ############################################
-        # There are 2 approaches, an empty database, or the same one that the app will use
-        #     create the tables
-        #     _db.create_all()
-        # or
-        # Use Alembic to load all of the DB revisions including supporting lookup data
-        # This is the path we'll use in legal_api!!
-
-        # even though this isn't referenced directly, it sets up the internal configs that upgrade needs
+        if database_exists(_db.engine.url):
```

Comment:
Quite a bit different than LEAR, I see LEAR doing alot of this, but it's easier to use the library I think